src/coreclr/jit/lower.cpp

```diff
@@ -7786,6 +7784,31 @@ void Lowering::TransformUnusedIndirection(GenTreeIndir* ind, Compiler* comp, Bas
     }
 }
 
+//------------------------------------------------------------------------
+// LowerIntrinsic: a common logic to lower INTRINSIC.
+//
+// Arguments:
+//    node - the INTRINSIC node we are lowering.
+//
+void Lowering::LowerIntrinsic(GenTreeIntrinsic* node)
+{
+    if (node->gtIntrinsicName == NI_System_Runtime_CompilerServices_RuntimeHelpers_IsKnownConstant)
+    {
+        // IsKnownConstant is expected to be folded in Importer/Morph/VN+ConstantProp
+        // This path is just in case if VN+ConstantProp didn't fold them for some reason (or e.g. they were disabled)
+        node->gtGetOp1()->SetUnusedValue();
+        LIR::Use use;
+        if (BlockRange().TryGetUse(node, &use))
+        {
+            use.ReplaceWith(comp->gtNewFalse());
+        }
+        BlockRange().Remove(node);
```

Comment:
This can just bash the node instead of searching for the use: `node->BashToConst(0);`
```suggestion
        node->BashToConst(0);
        return;
```
Without the `return`, `ContainCheckIntrinsic` below would see a removed node.